### DISCUSSION TITLE
How to use variable in global environment to render pdf in quarto

### DISCUSSION THREAD

**Discussion by joedoehl** [quarto-dev/quarto-cli#9096] [Q&A]
### Description

I am writing a report containing very data heavy analyses. Thus, when I want to render the pdf in quarto, I do not want to have to execute all code chunks every time, otherwise, it will take a day to render the document. Rmarkdown can use variables stored in the global environment, eliminating the need to execute the analysis code chunks every time I want to render the pdf to check that the format is alright. How can I get quarto to do the same? It seems quarto is set up to always initiate a new session in which it renders the document, which means, it cannot pull the variables stored in the global environment, which is quite a nuisance. Are there any workarounds for quarto?

**Comment by mcanouil** [comment]
I am not sure to understand what you are talking about, *i.e.*, the "global environment" thing.

If you were loading data into R then relying on the fact that `rmarkdown` is an R package to read what's in your session, then it is not possible in Quarto.
"`rmarkdown` runs in R" which is very different from "Quarto runs R".
Note that the way you are/were using `rmarkdown` is more of a side effect and in some aspect it is not considered a good practice.

This being said, you can cache/freeze results: https://quarto.org/docs/projects/code-execution.html#cache

You can also use R packages `targets` to efficiently manage your workflow and cache, then read from the cache to compute any report.
